pom.xml

```diff
@@ -108,10 +106,194 @@
     <mockito.version>4.9.0</mockito.version>
   </properties>
 
+  <profiles>
+
+    <profile>
+      <id>uPortal5</id>
+      <!-- designed for:
+        java 8
+        spring 4
+        tomcat <= 9.0.x
+      -->
+      <activation>
+        <activeByDefault>true</activeByDefault>
+      </activation>
+      <properties>
+        <target.java.version>1.8</target.java.version>
+        <tomcat.version>8.5.89</tomcat.version>
+        <spring.version>4.3.30.RELEASE</spring.version>
+        <!-- <portlet-api.version>2.0</portlet-api.version> -->
+      </properties>
+      <dependencies>
+        <!-- below is what is used by uP v5.17.0. -->
+        <dependency>
+          <groupId>org.apache.portals</groupId>
+          <artifactId>portlet-api_2.1.0_spec</artifactId>
+          <version>1.0</version>
+          <scope>test</scope>
+        </dependency>
+        <dependency>
+          <groupId>org.apache.portals.pluto</groupId>
+          <artifactId>pluto-taglib</artifactId>
+          <!-- does uP6 need v3.x of this? -->
+          <version>2.1.0-M3</version>
+          <scope>provided</scope>
+        </dependency>
+      </dependencies>
+    </profile>
+
+    <profile>
+      <id>uPortal6</id>
+      <!-- designed for:
+        java 11+
+        spring 5/6
+        tomcat 10.1+
+      -->
+      <activation>
+        <activeByDefault>false</activeByDefault>
+      </activation>
+      <properties>
+        <target.java.version>11</target.java.version>
+        <tomcat.version>10.1.34</tomcat.version>
+        <spring.version>5.3.28</spring.version>
+        <!-- <spring.version>6.2.2</spring.version> -->
+        <portlet-api.version>3.0</portlet-api.version>
+      </properties>
+      <dependencies>
+        <dependency>
+          <groupId>javax.portlet</groupId>
+          <artifactId>portlet-api</artifactId>
+          <version>${portlet-api.version}</version>
+          <scope>provided</scope>
+        </dependency>
+
+        <!-- Liferay's port of Spring portlets (needing for Spring 5+) -->
```

Comment:
Remove for uP5

Reply:
This line is contained w/in the uP6 profile. Check again?